include/maths/common/CSolvers.h

```diff
@@ -685,7 +698,7 @@ class MATHS_COMMON_EXPORT CSolvers {
             maxIterations -= n;
             return true;
         }
-        if (fa * fb > 0.0) {
+        if (std::signbit(fa) == std::signbit(fb)) {
```

Comment:
As above, this is slightly different to before if one or both variables is zero.

Reply:
Those conditions are actually handled in the code immediately above.